agent/consul/server_test.go

```diff
@@ -218,25 +216,16 @@ func testServerDCExpect(t *testing.T, dc string, expect int) (string, *Server) {
 	})
 }
 
-func testServerDCExpectNonVoter(t *testing.T, dc string, expect int) (string, *Server) {
-	return testServerWithConfig(t, func(c *Config) {
-		c.Datacenter = dc
-		c.Bootstrap = false
-		c.BootstrapExpect = expect
-		c.ReadReplica = true
-	})
-}
-
-func testServerWithConfig(t *testing.T, cb func(*Config)) (string, *Server) {
+func testServerWithConfig(t *testing.T, configOpts ...func(*Config)) (string, *Server) {
```

Comment:
By making this a varargs, I think we'll be able to:
* rename this to `testServer` since calling it with no `configOpts` functions is fine
* instead of separate `testServerX` methods, we can create `configOpts` functions and pass them in (ex: one for enabling ACLs, one for TLS, one for secondary). That allows us to compose the setup functions without wrapping the main function.


I'd like to also remove `newServer` and rename `testServer` to `newTestServer`, but I thought I'd do all of those renames at once in a separate PR, since it will change a few hundred lines.